File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int []dp;
    public int solve(int n){
        if(n<3){
            return n;
        }
        if(dp[n]!=0){
            return dp[n];
        }
        int takeOne=solve(n-1);
        int takeTwo=solve(n-2);
        dp[n]=takeOne+takeTwo;
        return dp[n];
    }
    public int climbStairs(int n) {
        dp=new int[n+1];
        return solve(n);
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        #this is use for memoization
        check=[0]*46
        
        def solve(rem,check):
            if rem<0:
                return 0
            if rem==0:
                return 1
            if check[rem] !=0:
                return check[rem]
            total=solve(rem-1,check)+solve(rem-2,check)
            #using memoization
            check[rem]=total
            return total
        
        return solve(n,check)
        